package.yaml

```diff
@@ -0,0 +1,44 @@
+name:                erd
+version:             0.1.3.0
+homepage:            https://github.com/BurntSushi/erd
+license:             PublicDomain
+license-file:        UNLICENSE
+author:              Andrew Gallant
+maintainer:          jamslam@gmail.com
+category:            Database, Development
+description:         Please see the README on GitHub at <https://github.com/BurntSushi/erd#readme>
+
+extra-source-files:
+  - README.md
+
+
+dependencies:
+  - base >= 4.5 && < 5
+  - graphviz 
+  - text 
+  - parsec 
+  - containers 
+  - bytestring 
+
+library:
+  source-dirs: src
+
+executables:
+  erd:
+    main: Main.hs
+    source-dirs: app
+    ghc-options:
+      - -Wall
+      - -fno-warn-name-shadowing
+      - -fno-warn-unused-do-bind
+    dependencies:
+      - erd
+
+tests:
+  spec:
+    main: Spec.hs
+    source-dirs:    test
+    dependencies:
+      - erd
+      - hspec
+      - QuickCheck
```

Comment:
Why are you adding a `package.yml` file? Please explain why we shouldn't just maintain the `erd.cabal` file directly.

Please try to improve the quality of commit messages so that they contain the rationale behind changes being made.

Reply:
@BurntSushi Hi, sorry I find I forgot to add a WIP tag in the PR. The PR is still on-working.  I will rebase the commits after it is done.

The reason of using package.yml is
1. I do not want to repeat the same dependencies in erd Main and tests
2. There are some boilerplate cabal I do not want to write it myself.  I would like to let hpack to handle it (in case there are some cabal changes in future, I will make the hpack automatically change it)

Thank you,
Jinxuan Zhu

Reply:
About cabal, it will come out automatically with `stack build`, so adding the package.yaml would not change the workflow.

Reply:
> 1. I do not want to repeat the same dependencies in erd lib, erd executes and tests

How many times do you need to repeat this? Is it once? If so, this does not seem like a compelling reason to switch because the dependency list should be updated very infrequently.

> 2\. There are some boilerplate cabal I do not want to write it myself.  I would like to let hpack to handle it

This doesn't sound like a good idea.

> About cabal, it will come out automatically with `stack build`, so adding the package.yaml would not change the workflow.

I think it does? It means folks using cabal will have a second class experience. Or has cabal been deprecated?

Please back up a bit and consider keeping things simple. Adding another layer of abstraction that looks virtually identical to the layer beneath it does not seem like a compelling enough advantage to me.

Reply:
@BurntSushi Agree. I  remove the package.yaml in 4c4f4ba.

Reply:
For the time being, hackage is an important source of haskell packages. To comply with the requirements of packaging - outlined: [hackage](http://hackage.haskell.org/upload) - it provides higher certainty when the cabal file is available explicitly. Until the duality of stack-cabal stays, cabal and stack as well shall be supported by the package.

Not even mentioning `cabal new`.